DOC: 2.x.rst - Add API REFERENCE link (#1721)
shapely/shapely: docs/release/2.x.rst

functions from the ``shapely.ops`` submodule such as ``polygonize()``) are
also made available in a vectorized version as top-level function.

> See ``API REFERENCE``: https://shapely.readthedocs.io/ for a full list of functions.

Reviewer: The link itself doesn't actually make much sense, since they already _are_ on that site (I suppose I forgot we don't have an actual link to the reference docs when I added that TODO link, we only have several pages under the "API REFERENCE" header in the sidebar). 

So maybe something like:

```suggestion
A full list of functions can be found in the API docs (see the pages listed under "API REFERENCE" in the left sidebar).
```

> A full list of functions can be found in the API docs (see the pages listed under "API REFERENCE" in the left sidebar).
> 

Reviewer: ```suggestion
A full list of functions can be found in the API docs (see the pages listed under "API REFERENCE" in the left sidebar).
```

> A full list of functions can be found in the API docs (see the pages listed under "API REFERENCE" in the left sidebar).
> 
> 

Reviewer: ```suggestion
A full list of functions can be found in the API docs (see the pages listed under "API REFERENCE" in the left sidebar).

```
